src/rustix.rs

```diff
@@ -57,6 +57,27 @@ pub fn socketpair_stream() -> io::Result<(SocketpairStream, SocketpairStream)> {
     })?)
 }
 
+/// Create a socketpair and return seqpacket handles connected to each end.
+#[inline]
+pub fn socketpair_seqpacket() -> io::Result<(SocketpairStream, SocketpairStream)> {
+    Ok(rustix::net::socketpair(
```

Comment:
I haven't seen this style before of "early Ok wrapping then `map()`".  What's the advantage over something like the following which I feel like I've seen much more commonly in other Rust code?

```
diff --git a/src/unix.rs b/src/unix.rs
index 9f8c876..ea58b29 100644
--- a/src/unix.rs
+++ b/src/unix.rs
@@ -55,22 +55,18 @@ pub fn socketpair_stream() -> io::Result<(SocketpairStream, SocketpairStream)> {
 /// Create a socketpair and return seqpacket handles connected to each end.
 #[inline]
 pub fn socketpair_seqpacket() -> io::Result<(SocketpairStream, SocketpairStream)> {
-    Ok(rustix::net::socketpair(
+    let (a, b) = rustix::net::socketpair(
         AddressFamily::UNIX,
         SocketType::SEQPACKET,
         SocketFlags::CLOEXEC,
         Protocol::default(),
-    )
-    .map(|(a, b)| {
-        let a = a.into_raw_fd();
-        let b = b.into_raw_fd();
-        unsafe {
-            (
-                SocketpairStream::from_raw_fd(a),
-                SocketpairStream::from_raw_fd(b),
-            )
-        }
-    })?)
+    )?;
+    unsafe {
+        Ok((
+            SocketpairStream::from_raw_fd(a.into_raw_fd()),
+            SocketpairStream::from_raw_fd(b.into_raw_fd()),
+        ))
+    }
 }
 
 impl Read for SocketpairStream {
```

Reply:
No particular advantage; it just fell out from how the code evolved one step at a time. I've changed it to your suggestion now, which I agree is easier to read :-).